### PR TITLE
Sup 7 customer create/edit

### PR DESCRIPTION
**Pull request by alexeyoverchenko**
bugfix

### CODE REVIEW

File: src/main/java/by/itech/lab/supplier/domain/Category.java [hunk 1]
```diff
@@ -18,7 +18,7 @@
 @NoArgsConstructor
 @Builder
 @Entity
-@Table(name = "category")
+@Table()
```
Comment (IrynaZazyba) [inline src/main/java/by/itech/lab/supplier/domain/Category.java]: remove brackets
```suggestion
@Table
```

File: src/main/java/by/itech/lab/supplier/domain/Customer.java [hunk 2]
```diff
@@ -21,7 +21,7 @@
 @NoArgsConstructor
 @Builder
 @Entity
-@Table
+@Table()
```
Comment (IrynaZazyba) [inline src/main/java/by/itech/lab/supplier/domain/Customer.java]: remove brackets

File: src/main/java/by/itech/lab/supplier/dto/mapper/CustomerMapper.java [hunk 3]
```diff
@@ -30,11 +30,17 @@ public CustomerDto map(Customer entity) {
         return CustomerDto.builder()
                 .id(entity.getId())
                 .name(entity.getName())
-                .registrationDate(entity.getRegistrationDate().toLocalDate())
+                .registrationDate(entity.getRegistrationDate())
                 .status(entity.isStatus())
                 .build();
     }
 
+    public void update(final CustomerDto from, final Customer to) {
```
Comment (IrynaZazyba) [inline src/main/java/by/itech/lab/supplier/dto/mapper/CustomerMapper.java]: method name unclear

File: src/main/java/by/itech/lab/supplier/service/impl/CustomerServiceImpl.java [hunk 4]
```diff
@@ -10,12 +13,17 @@
 import org.springframework.data.domain.Pageable;
 import org.springframework.stereotype.Service;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
 @Service
 @AllArgsConstructor
 public class CustomerServiceImpl implements CustomerService {
```
Comment (IrynaZazyba) [inline src/main/java/by/itech/lab/supplier/service/impl/CustomerServiceImpl.java:20]: use  BaseService interface

File: src/main/java/by/itech/lab/supplier/service/impl/CustomerServiceImpl.java [hunk 5]
```diff
@@ -29,4 +37,42 @@ public CustomerDto getCustomer(final Long customerId) {
...
+        }
+        return customers;
+    }
+
+    private UserDto createAdmin() {
+        return UserDto.builder()
+                .username("User name")
+                .name("Name")
+                .surname("Surname")
+                .email("email")
```
Comment (IrynaZazyba) [inline src/main/java/by/itech/lab/supplier/service/impl/CustomerServiceImpl.java]: u should set email value sent from frontend

File: src/main/java/by/itech/lab/supplier/service/impl/CustomerServiceImpl.java [hunk 6]
```diff
@@ -29,4 +37,42 @@ public CustomerDto getCustomer(final Long customerId) {
...
+                })
+                .orElseGet(() -> customerMapper.map(customerDto));
+
+        final Customer saved = customerRepository.save(customer);
+        userService.save(createAdmin());
+        return customerMapper.map(saved);
+    }
+
+    @Override
+    public List<Customer> saveNewStatus(List<CustomerDto> customerDtoList) {
```
Comment (IrynaZazyba) [inline src/main/java/by/itech/lab/supplier/service/impl/CustomerServiceImpl.java]: maybe change only status, not all entity

File: src/main/java/by/itech/lab/supplier/domain/Category.java (hunk 1 above)
Comment (alexeyoverchenko) [reply src/main/java/by/itech/lab/supplier/domain/Category.java]: done

File: src/main/java/by/itech/lab/supplier/domain/Customer.java (hunk 2 above)
Comment (alexeyoverchenko) [reply src/main/java/by/itech/lab/supplier/domain/Customer.java]: done

File: src/main/java/by/itech/lab/supplier/dto/mapper/CustomerMapper.java (hunk 3 above)
Comment (alexeyoverchenko) [reply src/main/java/by/itech/lab/supplier/dto/mapper/CustomerMapper.java]: How should I name it? "updateCustomer"? I suppose my name is obviously in this context.

File: src/main/java/by/itech/lab/supplier/service/impl/CustomerServiceImpl.java (hunk 4 above)
Comment (alexeyoverchenko) [reply src/main/java/by/itech/lab/supplier/service/impl/CustomerServiceImpl.java:20]: done

File: src/main/java/by/itech/lab/supplier/service/impl/CustomerServiceImpl.java (hunk 5 above)
Comment (alexeyoverchenko) [reply src/main/java/by/itech/lab/supplier/service/impl/CustomerServiceImpl.java]: It's not obligatory field while creating default admin, am I right? This field can be filled at the time of editing the admin, so I don't have to invent this as default

File: src/main/java/by/itech/lab/supplier/dto/mapper/CustomerMapper.java (hunk 3 above)
Comment (alexeyoverchenko) [reply src/main/java/by/itech/lab/supplier/dto/mapper/CustomerMapper.java]: changed to updateCustomer, cuz create updateStatus method in CustomerMapper

File: src/main/java/by/itech/lab/supplier/service/impl/CustomerServiceImpl.java (hunk 6 above)
Comment (alexeyoverchenko) [reply src/main/java/by/itech/lab/supplier/service/impl/CustomerServiceImpl.java]: good

File: src/main/java/by/itech/lab/supplier/service/impl/CustomerServiceImpl.java (hunk 5 above)
Comment (IrynaZazyba) [reply src/main/java/by/itech/lab/supplier/service/impl/CustomerServiceImpl.java]: 1. according to your code you fill email field in the user table with value 'email' - this creates an inconsistent state of the database, you should not think that someone will replace this 'email' with another really existed;
2. according to the specification Examples of workflows, Scenario1.  "The customer should provide an email address that is entered by the system admin in the form at the customer management page" etc.

File: src/main/java/by/itech/lab/supplier/dto/mapper/CustomerMapper.java (hunk 3 above)
Comment (IrynaZazyba) [reply src/main/java/by/itech/lab/supplier/dto/mapper/CustomerMapper.java]: it is a data transfer object, not a repository where you can update any entity, how can you update CustomerDto? I leave it up to you ...

**Pull request closed without merging**